Makefile

```diff
@@ -12,7 +12,17 @@ export GOPATH = $(GODEPS):$(USER_GOPATH)
 
 DEV_DEPS = github.com/uber/tchannel-go/thrift/thrift-gen \
 		   github.com/vektra/mockery/... \
-		   github.com/tools/godep
+		   github.com/tools/godep \
+		   github.com/golang/lint/golint
+
+# Automatically gather packages
+PKGS = $(shell find . -type d -maxdepth 3 \
+	! -path '*/.git*' \
+	! -path '*/_*' \
+	! -path '*/Godeps*' \
+	! -path '*/test*' \
+	! -path '*/examples*' \
+)
```

Comment:
Why not `/test*` and `/exmaples*`? Do we have a good reason for them to be not linted?